File: Data/Models/PayrollCodeMap.cs
namespace Blackwood.Access.Models
{
using System.ComponentModel.DataAnnotations;

    public class PayrollCodeMap
    {
        [Key]
        public int Id { get; set; }
        public int Type { get; set; }           // 0: Booking Type, 1: Availability Type
        public int TypeCode { get; set; }       // Booking or Availability Type Code
        public string Code { get; set; }        // Payroll Output Code
        public int? Unit { get; set; }           // Is paid as Unit - Deprecate?
        public bool? PayHours { get; set; }      // Hours count towards Actual Hours
        public bool? PayInstance { get; set;}    // Is paid as a single code instance
        public bool? ShiftCode { get; set; }     // Should be included in Shift calcs
        public bool? PayGaps { get; set; }       // Gaps between Bookings count towards Actual Hours
    }
}

File: ClientApp/app/models/PayrollCodeMap.ts
export class PayrollCodeMap {
    public id: number;
    public type: number;
    public typeCode: number;
    public code: string;
    public unit: boolean;
    public payHours: boolean;
    public payInstance: boolean;
    public shiftCode: boolean;
    public payGaps: boolean;
    public active: boolean;

    constructor(type, code) {
        this.type = type;
        this.typeCode = code;
        this.code = '';
        this.payHours = true;
        this.payInstance = false;
        this.shiftCode = false;
        this.payGaps = false;
        this.active = true;
    }
}